Etape 7

B-Evrard opened the pull request.

Commit:
Etape 7 : Passez un pays d'un écran à un autre

Commit:
Utilisation du ListViewModel dans Listview

Commit:
Séparation des 3 views de ListView en 3 fichiers et changement de nom ListView en TravelListView

valpacino commented on TravelVista/TravelVista/Views/ViewModels/TravelListViewModel.swift:
```diff
@@ -0,0 +1,24 @@
+
+import Foundation
+
+class TravelListViewModel: ObservableObject {
+    
+    
+    var regions: [Region] = Service().load("Source.json")
+    
+    
+}
```
Est-ce qu'on a vraiment besoin de ce ViewModel ? La vue pourrait juste prendre en input l'array de Region et ça marcherait tout aussi bien.
Après peut être que c'est OC qui te demande de fournir un ViewModel ?

B-Evrard replied:
C'est OC qui demande le ViewModel

valpacino commented on TravelVista/TravelVista/Views/SwiftUI/RegionListView.swift:
```diff
@@ -0,0 +1,29 @@
+    
+    var body: some View {
+        
+        Section(header:
+                    HStack {
```
Pense à l'indentation. Ça devrait être un réflexe

B-Evrard replied:
Effectivement, je corrige

valpacino commented on TravelVista/TravelVista/Views/SwiftUI/RegionListView.swift:
```diff
@@ -0,0 +1,29 @@
+
+import SwiftUI
+
+struct RegionListView: View {
```
Le mot List n'est pas approprié à mon sens ici. Ta vue représente une row d'une List. On pourrait l'appeler `RegionRowView` plutôt.

B-Evrard replied:
D'accord, je renomme également le fichier CountryView

B-Evrard committed:
Etape 7
Indentation code
Renomme RegionViewList et CountryView en RegionRowView et CountryRowView

valpacino commented on TravelVista/TravelVista/Views/SwiftUI/RegionRowView.swift:
```diff
@@ -0,0 +1,28 @@
+
+import SwiftUI
+
+struct RegionRowView: View {
```
RegionSectionView

B-Evrard committed:
Etape 7 - Renommer RegionRowView -> RegionSectionView
